master/master.cfg

```diff
@@ -401,26 +411,31 @@ def create_factory(os, llvm):
 
   if os.startswith('win'): return create_win_factory(os, llvm)
 
-  llvm_cmake_command = get_llvm_cmake_command(os)
-  env           = get_env(os)
+  config        = 'release'
+  env           = get_env(os, config)
   targets       = get_targets(os, llvm)
   make_threads  = get_make_threads(os)
 
   factory = BuildFactory()
 
   add_get_source_steps(factory, llvm)
 
-  factory.addStep(ShellCommand(name = 'Configure LLVM',
-                               description = 'Configure LLVM',
-                               locks = [performance_lock.access('counting')],
-                               workdir = 'llvm-build',
-                               env = env,
-                               haltOnFailure = True,
-                               command = llvm_cmake_command))
+  factory.addStep(CMake(name = 'Configure LLVM',
+                        description = 'Configure LLVM',
+                        locks = [performance_lock.access('counting')],
+                        haltOnFailure = True,
+                        env = env,
+                        workdir = 'llvm-build-%s' % config,
+                        path = '../llvm/',
+                        generator = get_cmake_generator(os),
+                        definitions = get_llvm_cmake_definitions(os, config),
+                        options = get_cmake_options(os),
+                        doStepIf=(lambda _: not isfile('llvm-build-%s/Makefile' % config))))
```

Comment:
So does this lambda run on the worker? I was very confused with how to use doStepIf

Reply:
well, I thought it did -- but either it doesn't, or I'm using it wrong, or something, but it doesn't seem to have any effect so far.

Reply:
Definitely doesn't run on the worker. Back to the drawing board.